cookbook/session/locale_sticky_session.rst

```diff
@@ -106,3 +106,86 @@ method::
     {
         $locale = $request->getLocale();
     }
+
+Setting the locale based on the user entity
+-------------------------------------------
+
+You might want to improve even further and want to define the locale based on
+the user entity of the logged in user. However since the `LocaleListener` is called
+before the `FirewallListener`, which is responsible for handling authentication and
+is setting the user token into the `TokenStorage`, you have no access to the user
+which is logged in.
+
+First lets pretend you have defined a property locale in your User Entity which you
+want to be used as the locale for the given user. In order to achieve the wanted locale
+configuration you can set the locale which is defined for the user to the session right
+after the login. Fortunately you can hook into the login process and update your session
+variable before the redirect to the first page. For this you need an event listener for the
+`security.interactive_login` event.
+
+    // src/AppBundle/EventListener/UserLocaleListener.php
+    namespace AppBundle\EventListener;
+
+    use Symfony\Component\HttpFoundation\Session\Session;
+    use Symfony\Component\Security\Http\Event\InteractiveLoginEvent;
+
+    /**
+     * Stores the locale of the user in the session after the
+     * login. This can be used by the LocaleListener afterwards.
+     */
+    class UserLocaleListener
+    {
+        /**
+         * @var Session
+         */
+        private $session;
+        public function __construct(Session $session)
+        {
+            $this->session = $session;
+        }
+        /**
+         * @param InteractiveLoginEvent $event
+         */
+        public function onInteractiveLogin(InteractiveLoginEvent $event)
+        {
+            $user = $event->getAuthenticationToken()->getUser();
+            $this->session->set('_locale', $user->getLocale());
```

Comment:
Can `$user->getLocale()` return a non-valid locale value? What would happen in that case?